app/controllers/answers_controller.rb

```diff
@@ -39,6 +39,7 @@ def destroy
 
   def accept
     return unauthorized_access unless @question.user == current_user
+    return invalid_request('Question has an already accepted answer') if @question.answers.any?(&:accepted)
```

Comment:
I will be making it into a `before_action` nevertheless _inclusive of the other validation in the action_